### PR TITLE
CB-12675: Allow force deletion of data lake

### PR DESCRIPTION
**Pull request by kmanamcheri**
Today, if any data hubs are attached, we block deletion of data lakes.
However, there are many use cases where we want to backup, delete,
recreate and restore on to a new datalake (with the same name). We
already have a --force option. Let's use this flag to force delete
a datalake when there are datahubs attached (and are stopped).

See detailed description in the commit message.

### CODE REVIEW

File: datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java [hunk 1]
```diff
@@ -852,8 +852,8 @@ private void checkIfSdxIsDeletable(SdxCluster sdxCluster, boolean forced) {
                 throw ex;
             }
         }
-        if (attachedDistroXClusters != null && !attachedDistroXClusters.isEmpty()) {
-            throw new BadRequestException(String.format("The following Data Hub cluster(s) must be terminated before SDX deletion [%s]",
+        if (!forced && attachedDistroXClusters != null && !attachedDistroXClusters.isEmpty()) {
+            throw new BadRequestException(String.format("The following Data Hub cluster(s) must be terminated before SDX deletion [%s]. Please use force delete to override this check.",
```
Comment (hreeve-cloudera) [inline datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java:856]: You mention in the JIRA, "We can check if the datahub is stopped, but we shouldn't block this operation." I think that'd be a good thing to check. Could we add that?

Comment (kkalvagadda1) [reply datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java:856]: I agree, we should check if the data hubs are stopped even when if it a forcible deletion.

Comment (kmanamcheri) [reply datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java:856]: Ok i'll add the check. So basically you can force this ONLY if the datahubs are stopped, right? Does that sound like the correct logic?

Comment (jamisonbennett) [reply datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java:856]: You might have situations where you can't stop datahubs due to a failure. I have a cluster where I can't stop FreeIPA and I have seen a datalake that couldn't be stopped too.

**Comment by kmanamcheri** [comment]
@hreeve-cloudera Added check for stopped datahubs

**Review by hreeve-cloudera — APPROVED** [review]

**Pull request merged**